api/src/main/java/me/lokka30/treasury/api/economy/transaction/EconomyTransaction.java

```diff
@@ -0,0 +1,227 @@
+/*
+ * This file is/was part of Treasury. To read more information about Treasury such as its licensing, see <https://github.com/lokka30/Treasury>.
+ */
+
+package me.lokka30.treasury.api.economy.transaction;
+
+import java.time.LocalDateTime;
+import java.util.Objects;
+import java.util.UUID;
+import me.lokka30.treasury.api.economy.currency.Currency;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Represents a transaction, made by an {@link me.lokka30.treasury.api.economy.account.Account}.
+ * A transaction is a move on a specific entity's balance, whether it is a {@link EconomyTransactionType#DEPOSIT} or
+ * {@link EconomyTransactionType#WITHDRAWAL}.
+ *
+ * @author lokka30, MrIvanPlays
+ * @since {@link me.lokka30.treasury.api.economy.misc.EconomyAPIVersion#v1_0 v1.0}
+ */
+public class EconomyTransaction {
+
+    /**
+     * Creates a new {@link EconomyTransaction.Builder}
+     *
+     * @return new builder
+     */
+    public static Builder newBuilder() {
+        return new Builder();
+    }
+
+    private final double transactionAmount;
+    @NotNull
+    private final UUID currencyUUID;
+    @NotNull
+    private final UUID initiatorUUID;
+    @NotNull
+    private final LocalDateTime timestamp;
+    @NotNull
+    private final EconomyTransactionType economyTransactionType;
+
+    /**
+     * Creates a new account transaction object.
+     *
+     * @param currencyUUID           the currency's UUID the transaction was made into
+     * @param initiatorUUID          the one who initiated the transaction
+     * @param economyTransactionType the transaction type
+     * @param transactionAmount      the amount which to deposit/withdraw
+     * @since {@link me.lokka30.treasury.api.economy.misc.EconomyAPIVersion#v1_0 v1.0}
+     */
+    public EconomyTransaction(
+            @NotNull UUID currencyUUID,
+            @NotNull UUID initiatorUUID,
+            @NotNull EconomyTransactionType economyTransactionType,
+            final double transactionAmount
+    ) {
+        this.currencyUUID = Objects.requireNonNull(currencyUUID, "currencyUUID");
+        this.initiatorUUID = Objects.requireNonNull(initiatorUUID, "initiatorUUID");
```

Comment:
What if the initiator is, for example, a vote plugin? Do we have to give an UUID to the vote plugin?

Reply:
ill make it nullable

Reply:
Not necessarily make it nullable
Maybe this idea is a bit over the top but, what if we have an interface `TransactionInitiator` and have several subclasses/subinterfaces like `PlayerTransactionInitiator`, `PluginTransactionInitiator` and `ServerTransactionInitiator` each one of these holding the information about the initiatior. The `PlayerTransactionInitiator` and UUID, the `PluginTransactionInitiator` the plugin name etc